### PR TITLE
CMR-9498 - Part 1 - Creating a redis hash cache.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: redis-utils-lib/test/cmr/redis_utils/test/test_redis_cache.clj [hunk 1]
```diff
@@ -38,3 +40,96 @@
...
+                        :native-format {:format :umm-json, :version "1.17.3"},
+                        :echo10 "<Collection><ShortName>SWOT_L2...",
+                        :dif "<DIF xmlns=\"http://gcmd.gsfc.n...",
+                        :dif10 "<DIF xmlns=\"http://gcmd.gsfc.n...",
+                        :iso19115 "<gmi:MI_Metadata xmlns:xs=\"htt...",
+                        :umm-json_1.17.3 "{  \"AdditionalAttributes\" : [ ..."}})
+
+(deftest redis-hash-cash-test
+  (let [cache-key :test-hash-cache
+        rhcache (redis-hash-cache/create-redis-hash-cache {:keys-to-track [cache-key]})]
```
Comment (jceaser) [inline redis-utils-lib/test/cmr/redis_utils/test/test_redis_cache.clj:76]: so you could create multiple caches in one use? I guess if your code needed to track two different things?

**Review by jceaser — APPROVED** [review]

**Review by TylerHeald1 — APPROVED** [review]

File: common-lib/src/cmr/common/hash_cache.clj [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   [cmr.common.cache :as cache]))
+
+(defn context->cache
+  "Get the cache for the given key from the context"
+  [context cache-key]
+  (cache/context->cache context cache-key))
+
+(defprotocol CmrHashCache
+  "Defines a protocol used for caching data using hashes."
+  (get-map
```
Comment (TylerHeald1) [inline common-lib/src/cmr/common/hash_cache.clj:13]: get-map and get-keys have the same variables, how are they going to differ? Would be good to have a comment for get-map

Comment (eereiter) [reply common-lib/src/cmr/common/hash_cache.clj:13]: Get map gets the entire hash map back, whereas get-keys only gets you the field names (hash keys) back.

Comment (eereiter) [reply common-lib/src/cmr/common/hash_cache.clj:13]: I will add a comment.

**Pull request merged**